app/models/partner.rb

```diff
@@ -360,4 +362,11 @@ def opening_times_is_json_or_nil
 
     errors.add :base, 'Partner.opening_times must be valid json'
   end
+
+  def three_or_less_category_tags
+    number_of_category_tags = tags.count { |t| t.type == 'Category' }
+    return if number_of_category_tags < 4
+
+    errors.add :base, 'Partner.tags can contain a maximum of 3 Category tags'
```

Comment:
Should we maybe also add some hint text in the UI so it's not a surprise? Or is that not precedent in this interface... like if it's not trivial to add or if it makes it look muddled, don't worry.

Reply:
It would probably be easy to add a little counter with stimulus to say something like "x/3 categories tags selected".

Is that the sort of thing you're thinking, or just like a paragraph next to the inputs warning that there's a limit?

Reply:
feel like this is a good idea but do we want to do too much UX work until the tag input is split into partnerships, categories and facilities tags? what we have now feels like a temp holding solution. if its holding us back i think we should prob do that first?

Reply:
![Screenshot 2023-05-01 at 10-53-36 Editing African Caribbean Care Group PlaceCal Admin](https://user-images.githubusercontent.com/6824891/235437556-7115d484-3962-4c51-b2af-10e9cbca358a.png)

Does this make it clear enough?